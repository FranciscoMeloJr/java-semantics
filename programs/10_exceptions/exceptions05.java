public class main {

  int x;

  int f(int y) {
    int t = 1;
    try{
      print(t," ");
      throw 5;
      print(8);      // not reachable
    } catch(int p) {
      print(p+10," ");
    }
    for (int i = 1; i<=y; ++i)
      t = t*i;
    return t;
  }

  void main(string[] args) {
    x = 5;
    print(f(x),"\n");
    print("Done!","\n");
  }
}

// 1 15 120
// Done!
