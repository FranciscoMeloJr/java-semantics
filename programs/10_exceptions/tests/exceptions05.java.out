1 15 120
Done!
